email-service/internal/handlers/subscriber.go

```diff
@@ -0,0 +1,60 @@
+package handlers
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/models"
+)
+
+var logger *slog.Logger
+
+func getLogger() *slog.Logger {
+	if logger == nil {
+		logger = slog.Default().With(slog.String("src", "handler"))
+	}
+	return logger
+}
+
+type subscriberRepository interface {
+	Create(subscriber *models.Subscriber) error
+	Delete(subscriber *models.Subscriber) error
+}
+
+type SubscriberEvents struct {
+	repo subscriberRepository
+}
+
+func (s *SubscriberEvents) Subscribe(ctx context.Context, email string) error {
+	getLogger().Info("new subscriber")
```

Comment:
why are you calling `getLogger` each time instead of calling in once? 
```go
logger := getLogger()
logger.Info("saving subscriber")
logger.Info("subscriber saved")
```